src/SuperGoodRobot/SuperGoodRobot.ino

```diff
@@ -0,0 +1,447 @@
+/* ENPH 253 PRELIMINARY CODE
+last updated: 070815
+version: 0.2
+
+*/
+
+//============== LIBRARIES ==============
+#include <phys253.h>
+#include <LiquidCrystal.h>
+//#include <Servo253.h>
+#include <avr/interrupt.h>
+
+#include <EEPROMex.h>
+#include <EEPROMVar.h>
+
+#include "TAPE_FOLLOWING.h"
+
+//============== PINS ==============
+//ANALOG IO
+#define TAPE_SENSOR_LEFT 0
+#define TAPE_SENSOR_RIGHT 1
+#define IR_FRONT_1 2
+#define IR_FRONT_2 3
+#define IR_SIDE 4
+#define TAPE_SENSOR_SIDE 5
+#define TAPE_SENSOR_FRONT 7
+
+//SERVOS
+#define SERVO_FOREARM 0
+#define SERVO_BASEARM 1
+#define SERVO_BASE 2
+
+//MOTORS
+#define MOTOR_LEFT 0
+#define MOTOR_RIGHT 1
+#define MOTOR_CLAW 2
+
+//DIGITAL IO
+//INPUT
+#define SWITCH_CLAW_OPEN 4
+#define SWITCH_CLAW_CLOSE 5
+#define SWITCH_PLUSH_DETECT 6
+#define HOOK_ARM_TOUCH_ON 7
+#define HOOK_ARM_TOUCH_OFF 8
+
+//OUTPUT
+#define HOOK_ARM_RETRACT 15
+#define HOOK_ARM_SPRING_UP 14
+#define INTAKE_ROLLER_MOTOR 12
+
+//============== DRIVE SETTINGS ==============
+//QRD DRIVE
+#define QRD_THRESHOLD 200
+
+//IR DRIVE
+
+//ARM CONSTANTS
+
+//SERVO POSITIONS
+#define BASESERVO_POSITION_START 90//some number
+#define FOREARMSERVO_POSITION_START 90//some number
+#define BASE_POSITION_START 90//some number
+
+//FLAGS
+boolean upRamp=false;
+boolean sideTapePresent=false;
+boolean clawOpen=false;
+boolean previousSwitchVal=true;
+
+#define KNOB_MAX 1023
+#define STANDARD_DELAY_1 500 //ms
+#define STANDARD_DELAY_2 200 //ms for faster things
+//============== STATES ==============
+enum RobotState {
+	INITIALISING, PAST_DOOR, FOLLOW_TAPE_1, COLLECT_ITEM_1, FOLLOW_TAPE_2, COLLECT_ITEM_2, COLLECT_ITEM_3, COLLECT_ITEM_4, COLLECT_ITEM_5, COLLECT_ITEM_6, UP_RAMP, PAST_RAMP, IR, ZIPLINE, FINISHED, TEST,DRIVE, CLAW_ARM_TEST, MAIN_MENU, STATE_MENU, TAPE_MENU, NUM_STATES
+};
+enum Plan { 
+	FULL, NOT4, NOT5, TEST_DRIVE, NUM_PLAN //placeholder for lack of len(enum) function, don't delete
+};
+
+//SET STATES
+RobotState currentState = INITIALISING;
+RobotState lastState = INITIALISING;  
+
+Plan currentPlan=FULL;
+
+//============== INTERRUPTS ==============
+/*volatile unsigned int INT_0 = 0;
+ISR(INT0_vect) {LCD.clear(); LCD.home(); LCD.print("INT0: "); LCD.print(INT_0++);};
+ISR(INT0_vect) {
+  switchState(MAIN_MENU);
+}*/
+
+//============== SETUP ==============
+void setup() {
+	#include <phys253setup.txt>  
+	portMode(0, INPUT) ;
+	portMode(1, OUTPUT) ;
+
+	LCD.clear();
+	LCD.home();
+
+	RCServo0.attach(RCServo0Output) ;
+	RCServo1.attach(RCServo1Output) ;
+	RCServo2.attach(RCServo2Output) ;
```

Comment:
Are these necessary?  It's my understanding that in the phys253.h library this is already done.